pandas/_libs/tslibs/arraylike.pyx

```diff
@@ -0,0 +1,717 @@
+# -*- coding: utf-8 -*-
+
```

Comment:
if you end up creating this file, can you take the opportunity of the refactor to add a module docstring here to better document what / what type of functionality is included in this file? 

But I agree with Jeff that from a quick view some of those functions seems to be about parsing / conversion?